Normalizer/ProductValueNormalizer.php

```diff
@@ -45,14 +62,62 @@ public function normalize($value, $format = null, array $context = [])
             $data['scope'] = $value->getScope();
         }
 
-        $backendType = $value->getAttribute()->getBackendType();
+        $attribute   = $value->getAttribute();
+        $backendType = $attribute->getBackendType();
+        $key         = $this->getKeyForValue($value, $attribute, $backendType);
+        $data[$key]  = $this->normalizeValueData($value->getData(), $backendType, $context);
 
-        if ('options' !== $backendType) {
-            $data[$backendType] = $this->normalizeValueData($value->getData(), $backendType, $context);
-        } else {
-            $data['optionIds'] = $this->normalizeValueData($value->getData(), $backendType, $context);
+        return $data;
+    }
+
+    /**
+     * Decide what is the key used for data inside the normalized product value
+     *
+     * @param ProductValueInterface $value
+     * @param AttributeInterface    $attribute
+     * @param string                $backendType
+     *
+     * @return string
+     */
+    protected function getKeyForValue(ProductValueInterface $value, AttributeInterface $attribute, $backendType)
+    {
+        if ('options' === $backendType) {
+            return 'optionIds';
         }
 
-        return $data;
+        $refDataName = $attribute->getReferenceDataName();
+        if (null !== $refDataName) {
+            if ('reference_data_options' === $backendType) {
```

Comment:
Nope, this case is only for multiple reference data and assets.
For simple reference data (for which backend type is "reference_data_option", singular) the key used is the reference data name (see just below).